crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -0,0 +1,329 @@
+/// parameters 'sticks' for future requests until a new filter overwrites them.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct SyncRequestListFilters {
+    // These fields may be expanded through use of extensions.
+    /// Whether to return DMs, non-DM rooms or both.
+    ///
+    /// Flag which only returns rooms present (or not) in the DM section of account data.
+    /// If unset, both DM rooms and non-DM rooms are returned. If false, only non-DM rooms
+    /// are returned. If true, only DM rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_dm: Option<bool>,
+
+    /// Only list rooms that are spaces of these or all
+    ///
+    /// A list of spaces which target rooms must be a part of. For every invited/joined
+    /// room for this user, ensure that there is a parent space event which is in this list. If
+    /// unset, all rooms are included. Servers MUST NOT navigate subspaces. It is up to the
+    /// client to give a complete list of spaces to navigate. Only rooms directly in these
+    /// spaces will be returned.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub spaces: Vec<String>,
+
+    /// Whether to return encrypted, non-encrypted rooms or both.
+    ///
+    /// Flag which only returns rooms which have an `m.room.encryption` state event. If
+    /// unset, both encrypted and unencrypted rooms are returned. If false, only unencrypted
+    /// rooms are returned. If true, only encrypted rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_encrypted: Option<bool>,
+
+    /// Whether to return invited Rooms, only joined rooms or both.
+    ///
+    /// Flag which only returns rooms the user is currently invited to. If unset, both
+    /// invited and joined rooms are returned. If false, no invited rooms are returned. If
+    /// true, only invited rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_invite: Option<bool>,
+
+    /// Whether to return Rooms with tombstones, only rooms without tombstones or both.
+    ///
+    /// Flag which only returns rooms which have an `m.room.tombstone` state event. If unset,
+    /// both tombstoned and un-tombstoned rooms are returned. If false, only un-tombstoned rooms
+    /// are returned. If true, only tombstoned rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_tombstoned: Option<bool>,
+
+    /// Only list rooms of given create-types or all
+    ///
+    /// If specified, only rooms where the `m.room.create` event has a `type` matching one
+    /// of the strings in this array will be returned. If this field is unset, all rooms are
+    /// returned regardless of type. This can be used to get the initial set of spaces for an
+    /// account.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub room_types: Vec<String>,
```

Comment:
In other places, a filter of `room_types: []` is semantically different from not specifying `room_types`. Are you sure it's not the same here? If it *is* the same, this has to be `Option<Vec<String>>` to distinguish these two cases.

Reply:
no, this only really only makes sense as existing and with items, or unset. An empty list would mean no item would ever match, which doesn't make sense and isn't how it works (and a potential foot gun even)